snuba/query/snql/expression_visitor.py

```diff
@@ -136,10 +138,13 @@ def visit_numeric_literal(node: Node, visited_children: Iterable[Any]) -> Litera
         return Literal(None, float(node.text))
 
 
+newline_re = re.compile("(?:\\{2})*(\\n)")
+
+
 def visit_quoted_literal(node: Node, visited_children: Tuple[Any]) -> Literal:
-    # In order to match any character except for \' there is a crazy regex that needs to be
-    # stripped and de-escaped.
-    match = node.text[1:-1].replace("\\'", "'")
+    text = node.text[1:-1]
+    text = newline_re.sub(text, "\n")
+    match = text.replace("\\'", "'").replace("\\\\", "\\")
```

Comment:
I think this is going to transform `\\\\\n  with \n` instead of `\\\n`

Reply:
You're right. I'll have to rethink all of this.

Reply:
So after adding numerous tests for this, the test cases suggest this works as expected, even if it's not intuitive why.